uaclient/files/notices.py

```diff
@@ -0,0 +1,153 @@
+import os
+from collections import namedtuple
+from enum import Enum
+from typing import Optional
+
+from uaclient import event_logger, exceptions, system
+from uaclient.files import UAFile
+
+event = event_logger.get_event_logger()
+NoticeFileDetails = namedtuple("NoticeFileDetails", ["order_id", "label"])
+
+
+class NoticesLabelID(NoticeFileDetails, Enum):
+    REBOOT_REQUIRED = NoticeFileDetails(label="reboot_required", order_id="10")
+    ENABLE_REBOOT_REQUIRED = NoticeFileDetails(
+        label="enable_reboot_required", order_id="11"
+    )
+    REBOOT_SCRIPT_FAILED = NoticeFileDetails(
+        label="reboot_script_failed", order_id="12"
+    )
+    FIPS_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_reboot_required", order_id="20"
+    )
+    FIPS_SYSTEM_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_system_reboot_required", order_id="21"
+    )
+    FIPS_INSTALL_OUT_OF_DATE = NoticeFileDetails(
+        label="fips_install_out_of_date", order_id="22"
+    )
+    FIPS_DISABLE_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_disable_reboot_required", order_id="23"
+    )
+    FIPS_PROC_FILE_ERROR = NoticeFileDetails(
+        label="fips_proc_file_error", order_id="24"
+    )
+    FIPS_MANUAL_DISABLE_URL = NoticeFileDetails(
+        label="fips_manual_disable_url", order_id="25"
+    )
+    WRONG_FIPS_METAPACKAGE_ON_CLOUD = NoticeFileDetails(
+        label="wrong_fips_metapackage_on_cloud", order_id="25"
+    )
+    LIVEPATCH_LTS_REBOOT_REQUIRED = NoticeFileDetails(
+        label="lp_lts_reboot_required", order_id="30"
+    )
+    CONTRACT_REFRESH_WARNING = NoticeFileDetails(
+        label="contract_refresh_warning", order_id="40"
+    )
+    DAEMON_AUTO_ATTACH_LOCK_HELD = NoticeFileDetails(
+        label="daemon_auto_attach_lock_held", order_id="50"
+    )
+    DAEMON_AUTO_ATTACH_FAILED = NoticeFileDetails(
+        label="daemon_auto_attach_failed", order_id="51"
+    )
+    OPERATION_IN_PROGRESS = NoticeFileDetails(
+        label="operation_in_progress", order_id="60"
+    )
+
+
+class Notices:
+    def __init__(
+        self,
+        is_permanent: bool = True,
+        root_mode: bool = True,
+        override_dir: Optional[str] = None,
+    ):
+        if not override_dir:
+            self.directory = "/var/lib/ubuntu-advantage/notices/"
+            if not is_permanent:
+                self.directory = "/var/run/notices/"
```

Comment:
Let's move those paths to the defaults module

Reply:
Also, instead of having to pass the `is_permanent` variable to dictate that the notice should not be on `/var/run`, we could instead add a parameter `NoticeFileDetails` for that, since we know that all reboot notice should have the same behavior, we can control that directly through the `NoticeFileDetails` instance